server/src/main/java/org/opensearch/index/store/RemoteDirectoryFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store;
+
+import org.apache.lucene.store.Directory;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.shard.ShardPath;
+import org.opensearch.plugins.IndexStorePlugin;
+import org.opensearch.repositories.Repository;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+
+import java.io.IOException;
+
+/**
+ * Factory for a remote store directory
+ *
+ * @opensearch.internal
+ */
+public class RemoteDirectoryFactory implements IndexStorePlugin.RemoteDirectoryFactory {
```

Comment:
Why is this internal when the RemoteDirectory class is flagged `@opensearch.api`? Have we clearly documented what can be (if anything) overriden by third party plugins? If so we should update the javadocs here to include that documentation otherwise we risk plugins recklessly overriding core internals.

Reply:
`RemoteDirectory` class should also be marked as `@opensearch.internal`, will make the change.